Chord chart connectivity

frheault opened the pull request:
Added options for visualization of connectivity, chord chart!
This is very hard to do in matplotlib (or even in fancy library like Plotly), I used some code I found online and modified it to our need.

This option works best with matrix containing around 25-250 connections, more than that and it requires heavy fine-tunning for it to look nice (likely not possible with the scripts).

It also has some difficulties if the biggest connection is 10x higher values than the second or weird extreme distribution.
Otherwise it works pretty well.

Only makes sense for streamlines count and volume matrix, does not make sense with length or FA.

![sc_vol_normalized_chord-min](https://user-images.githubusercontent.com/10820351/95882695-ff2e5e00-0d47-11eb-83fd-bb4dee8becfa.png)

https://drive.google.com/file/d/1Zka8-xmb1xwJGT7QRW4NRO3LaQIkk3W6/view?usp=sharing

pep8speaks commented:
Hello @frheault, Thank you for submitting the Pull Request !



* In the file [`scripts/scil_visualize_connectivity.py`](https://github.com/scilus/scilpy/blob/fd15b3b610c339ff33c9d5aa30ce3e57a1d97412/scripts/scil_visualize_connectivity.py):

> [Line 83:80](https://github.com/scilus/scilpy/blob/fd15b3b610c339ff33c9d5aa30ce3e57a1d97412/scripts/scil_visualize_connectivity.py#L83): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 92:80](https://github.com/scilus/scilpy/blob/fd15b3b610c339ff33c9d5aa30ce3e57a1d97412/scripts/scil_visualize_connectivity.py#L92): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)

arnaudbore commented on scilpy/viz/chord_chart.py:
```diff
@@ -0,0 +1,235 @@
+
+
+def polar2xy(r, theta):
+    return np.array([r*np.cos(theta), r*np.sin(theta)])
```
Please add/update docstring parameters and return

arnaudbore reviewed (commented):
Fancy visualisation !
Please add docstring.
There is some issues with your example (some files are missing). 
Something I may be missing, the histogram is weird to me. What you called normalised histogram should it be between 0 and 1 ? Is there a way to add some legend as well ?

mdesco commented on scripts/scil_visualize_connectivity.py:
```diff
@@ -74,6 +78,25 @@ def _build_arg_parser():
+                       help='Compute and display/save a chord chart of weigth.')
+    chord.add_argument('--percentile_threshold', type=int, default=0,
+                       help='Discard connections below that percentile.'
+                            '[%(default)s]')
```
Not sure what this percentile represents...

mdesco reviewed (approved):
Once minor comments addressed, everything looks good.

frheault commented:
@arnaudbore I added the docstring to the simple function (polar coord) that could be useful to someone else.

I didn't add docstring to the very specific function borrowed from the net, Laurent would need that script ASAP (I slowed him down because I moved)

arnaudbore reviewed (approved):
Go LGTM !
